library/std/src/sys/pal/uefi/helpers.rs

```diff
@@ -291,6 +291,13 @@ impl Drop for DevicePath {
     }
 }
 
+impl crate::fmt::Debug for OwnedDevicePath {
+    fn fmt(&self, f: &mut crate::fmt::Formatter<'_>) -> crate::fmt::Result {
+        let p = device_path_to_text(self.0).unwrap();
```

Comment:
Can this `unwrap` fail? Wondering if it would be better to not `unwrap` and instead print the raw bytes or something if `device_path_to_text` fails.

Reply:
It can fail if `EFI_DEVICE_PATH_TO_TEXT_PROTOCOL` is not present. It should be present in all UEFI 2.0+ systems, but well probably better not to have that limitation.
However, I do think it would be better to return Error, instead of printing the raw bytes in that case.

I can add `DevicePathNodeIterator` and `DevicePathNode` to walk and print each node data in a specific format, but well, the Debug implementation will end up being wildly different in that case.

Reply:
Ok, so looking at the docs of `fmt::Error`, it seems like it should never be constructed by implementors. So, I guess walking the DevicePath and printing each member's details is the way to go.

Reply:
Unfortunately that makes the PR quite a bit more complicated; I'd really like to move towards smaller, very focused PRs for the UEFI changes. How about if we go back to `device_path_to_text(self.0)`, and in the failure case just do something simple like `f.debug_struct("OwnedDevicePath").finish_non_exhaustive()`? That should provide useful output 99% of the time, and keep the 1% case simple.